src/model.ts

```diff
@@ -1,4 +1,88 @@
 import WebSocket = require("ws");
+import TiciTacaToeyGameEngine from "./TiciTacaToeyGameEngine";
+
+interface TimerBase {
+  isRunning: boolean;
+  timeLeft: number;
+  increment: number;
+}
+
+class Timer implements TimerBase {
+  isRunning: boolean;
+  #startTime: number;
+  timeLeft: number;
+  #intervalID;
+  #playerId: string;
+  #gameId: string;
+  increment: number;
+
+  constructor(
+    allotedTime: number,
+    playerId: string,
+    gameId: string,
+    increment: number
+  ) {
+    this.#playerId = playerId;
+    this.#gameId = gameId;
+    this.reset(allotedTime, increment);
+  }
+
+  #getTimeElapsedSinceLastStart() {
+    return this.#startTime === 0 ? 0 : Date.now() - this.#startTime;
+  }
+
+  start(engine: TiciTacaToeyGameEngine) {
+    if (this.isRunning) {
+      return "Timer is already running";
+    }
+
+    const playerTimeoutMessage: PlayerTimeoutMessage = {
+      type: MessageTypes.PLAYER_TIMEOUT,
+      gameId: this.#gameId,
+      playerId: this.#playerId,
+    };
+
+    this.isRunning = true;
+    this.#startTime = Date.now();
+
+    this.#intervalID = setInterval(() => {
+      this.timeLeft = this.timeLeft - this.#getTimeElapsedSinceLastStart();
+      this.#startTime = Date.now();
+      console.log(this.timeLeft);
+      if (this.timeLeft <= 0) {
+        engine.play(playerTimeoutMessage);
+        this.stop();
+        return;
+      }
+      const timeUpdateMessage: UpdateTimeMessage = {
+        type: MessageTypes.NOTIFY_TIME,
+        gameId: this.#gameId,
+      };
+      engine.play(timeUpdateMessage); // notifying the client every 100 ms
+    }, 100);
+  }
+
+  stop() {
```

Comment:
Instead of the `reset` and `constructor` methods can we just accept a parameter in the `stop` method called `increment` which adds the increment to timeLeft?